include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -8146,73 +7433,25 @@ FEFaceEvaluation<dim,
-  if (fe_degree > -1 &&
-      this->subface_index >= GeometryInfo<dim - 1>::max_children_per_cell &&
-      this->data->element_type <=
-        internal::MatrixFreeFunctions::tensor_symmetric)
-    internal::FEFaceEvaluationImpl<
-      true,
-      dim,
-      fe_degree,
-      n_q_points_1d_actual,
-      n_components,
-      VectorizedArrayType>::integrate_in_face(*this->data,
-                                              temp1,
-                                              this->begin_values(),
-                                              this->begin_gradients(),
-                                              this->scratch_data +
-                                                2 * n_components *
-                                                  dofs_per_face,
-                                              integrate_values,
-                                              integrate_gradients,
-                                              this->subface_index);
-  else
-    internal::FEFaceEvaluationImpl<
-      false,
-      dim,
-      fe_degree,
-      n_q_points_1d_actual,
-      n_components,
-      VectorizedArrayType>::integrate_in_face(*this->data,
-                                              temp1,
-                                              this->begin_values(),
-                                              this->begin_gradients(),
-                                              this->scratch_data +
-                                                2 * n_components *
-                                                  dofs_per_face,
-                                              integrate_values,
-                                              integrate_gradients,
-                                              this->subface_index);
-
-  internal::FEFaceNormalEvaluationImpl<dim,
-                                       fe_degree,
-                                       n_components,
-                                       VectorizedArrayType>::
-    template interpolate<false, false>(
-      *this->data, temp1, values_array, integrate_gradients, this->face_no);
+  internal::FEFaceEvaluationSelector<
+    dim,
+    fe_degree,
+    n_q_points_1d,
+    n_components,
+    Number,
+    VectorizedArrayType>::integrate(*this->data,
+                                    values_array,
+                                    this->begin_values(),
+                                    this->begin_gradients(),
+                                    this->scratch_data,
+                                    integrate_values,
+                                    integrate_gradients,
+                                    this->face_no,
+                                    this->subface_index,
+                                    this->face_orientation,
+                                    this->mapping_data
+                                      ->descriptor[this->active_fe_index]
+                                      .face_orientations);
```

Comment:
Here is one spot of the simpler kind I moved to a single implementation file. This is similar to `SelectEvaluator` in `FEEvaluation` but now for the face variant. If you @masterleinad want I can pick a name a bit closer to the cell selection class. I don't know, maybe `SelectEvaluatorFace` maybe. (But in the present we do not have any of the recursive parts we could do for the cell part).

Reply:
This is an internal class anyway so we don't need to overly picky with the name. I am fine with that.